clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -153,4 +154,16 @@ public class AzureComputeClient extends AzureBaseClient {
     }
     null
   }
+
+  ServiceResponse<Void> destroyServerGroup(String resourceGroupName, String serverGroupName) {
+    try {
+      this.client.getVirtualMachineScaleSetsOperations()?.delete(resourceGroupName, serverGroupName)
+    } catch (CloudException e) {
+      // treat exception as a http 404 return (resource not found)
+      log.warn("ServerGroup: ${e.message} (${serverGroupName} was not found?)")
+    }
```

Comment:
The issue here is with decoding the CloudException to get the real reason for the failure; if the error is a "timeout", we could retry the operation. The delete server group is taking some significant time (I think the instances must be shut down first). 

I will modify this and throw a RuntimeException which we can use to retry on (five times?).